packages/core/src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -44,12 +52,70 @@ function setup(options: ReactNativeErrorHandlersOptions): void {
  * Setup unhandled promise rejection tracking
  */
 function setupUnhandledRejectionsTracking(patchGlobalPromise: boolean): void {
-  if (patchGlobalPromise) {
-    polyfillPromise();
-  }
+  try {
+    if (isHermesEnabled() && RN_GLOBAL_OBJ.HermesInternal?.enablePromiseRejectionTracker) {
+      logger.log('Using Hermes native promise rejection tracking');
+
+      RN_GLOBAL_OBJ.HermesInternal.enablePromiseRejectionTracker({
+        allRejections: true,
+        onUnhandled: (id: string, error: unknown) => {
+          if (__DEV__) {
+            // eslint-disable-next-line no-console
+            console.warn(`Possible Unhandled Promise Rejection (id: ${id}):\n${error}`);
+          }
+
+          captureException(error, {
+            data: { id },
+            originalException: error,
+            syntheticException: isErrorLike(error) ? undefined : createSyntheticError(),
+            mechanism: { handled: true, type: 'onunhandledrejection' },
```

Comment:
why handled true here?

Reply:
Good question @lucas-zimerman 👍 
I kept [the same `captureException` call from the existing `attachUnhandledRejectionHandler` below](https://github.com/getsentry/sentry-react-native/blob/11f67eb3cf750fae2a7182d0948450b6fb288791/packages/core/src/js/integrations/reactnativeerrorhandlers.ts#L84).

Reply:
For reference https://github.com/getsentry/sentry-react-native/issues/4141
And https://github.com/getsentry/sentry-react-native/pull/4457#discussion_r1922179466
```
We have to mark the rejection handled: true to avoid breaking crash rate calculations.

Rejections do not crash the app in RN at the moment.
```
Might be worth adding a small comment in the comment pointing to https://github.com/getsentry/sentry-react-native/issues/4141

Reply:
Thank you for the context @lucas-zimerman 🙇 
I didn't know that. I've added comments with 92ae70abd4b415f60c72dd8b60a234e5c208e4de